bin/API_Migration_scripts_readme/WebScraper_Java_Code.ipynb

```diff
@@ -0,0 +1,376 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "361542ff",
+   "metadata": {},
+   "source": [
+    "# Python Scripts - Text extraction (webscraping) and matching contents of Java file(s)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8a2f0458",
+   "metadata": {},
+   "source": [
+    "## Created for API Migration"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ab88f421",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#imports\n",
+    "import os\n",
+    "import csv\n",
+    "import requests\n",
+    "from bs4 import bs4\n",
+    "import csv"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "aaa1ab25",
+   "metadata": {},
+   "source": [
+    "## >>>WebScraper - will save a CSV and a JSON file --JSON not required--<<<"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a251d7fc",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "url = \"https://developer.android.com/sdk/api_diff/33/changes/alldiffs_index_changes\" #URL from android website - only changes\n",
+    "\n",
+    "#Request and creation of Beautiful Soup with the response\n",
+    "response = requests.get(url)\n",
+    "soup = BeautifulSoup(response.content, \"html.parser\")\n",
+    "\n",
+    "#getting links from the webpage\n",
+    "links = []\n",
+    "for a in soup.find_all(\"a\"):\n",
+    "    if a.has_attr(\"href\") and a[\"href\"].startswith(\"/sdk/api_diff/33/changes/\"):\n",
+    "        links.append(a[\"href\"])\n",
+    "\n",
+    "#creation and writing the file with DictWriter\n",
+    "csv_file = open(\"classes.csv\", \"w\", newline=\"\")\n",
+    "csv_writer = csv.DictWriter(csv_file, fieldnames=[\"Package Class\"])\n",
+    "csv_writer.writeheader()\n",
+    "\n",
+    "#technically not writing any links but the classes themselves\n",
+    "for link in links:\n",
+    "    css_class = link.split('/')[-1]\n",
+    "    csv_writer.writerow({\"Package Class\": css_class})\n",
+    "\n",
+    "\n",
+    "#JSON not required\n",
+    "json_file = open(\"classes.json\", \"w\")\n",
+    "json_data = {\"links\": links}\n",
+    "json.dump(json_data, json_file)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "355d9124",
+   "metadata": {},
+   "source": [
+    "## python script to compare the CSV against Java files in a given directory\n",
+    "##### python script to use the obtained CSV file. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2e458972",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def match_csv_java(csv_file, directory): #method accepting CSV and a directory\n",
+    "    java_files = os.listdir(directory)\n",
+    "    with open(csv_file, \"r\") as f: # open CSV in read mode\n",
+    "        reader = csv.reader(f)\n",
+    "        for row in reader:\n",
```

Comment:
why do we have two separate `match_csv_java` files which appear to be copy-pasted duplicates of each other? Remember the DRY principle